### PR TITLE
refactor(number-input): migrate to TypeScript

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- blocked by #1867
- This is cherry picked, so I expect that the build will fail until the blocking PR is merged

### CODE REVIEW

File: packages/components/inputs/number-input/src/number-input.tsx [hunk 1]
```diff
@@ -151,13 +149,13 @@ NumberInput.toFormValue = (numberOrString) => {
...
 
-NumberInput.isEmpty = (value) => {
+NumberInput.isEmpty = (value: number | string): boolean => {
   if (typeof value === 'string') return value.trim().length === 0;
   if (typeof value === 'number') return isNaN(value);
   return true;
 };
 
-NumberInput.hasFractionDigits = (number) => {
+NumberInput.hasFractionDigits = (number: number) => {
```
Comment (dogayuksel) [inline packages/components/inputs/number-input/src/number-input.tsx:158]: warning message sounds like i may be able to pass in a string type to this function, but the function signature only specifies numbers.

Comment (adnasa) [reply packages/components/inputs/number-input/src/number-input.tsx:158]: that's my mistake. good catch!

Comment (adnasa) [reply packages/components/inputs/number-input/src/number-input.tsx:158]: that's my mistake, and good catch on the message.

**Review by tdeekens — APPROVED** [review]
💯

**Pull request merged**